controllers/servers/host_definer/kubernetes_manager/manager.py

```diff
@@ -0,0 +1,265 @@
+
+logger = get_stdout_logger()
+
+
+class KubernetesManager():
+    def __init__(self):
+        self._load_cluster_configuration()
+        self.dynamic_client = self._get_dynamic_client()
+        self.storage_api = client.StorageV1Api()
+        self.core_api = client.CoreV1Api()
+        self.custom_object_api = client.CustomObjectsApi()
+        self.apps_api = client.AppsV1Api()
+        self.csi_nodes_api = self._get_csi_nodes_api()
+        self.host_definitions_api = self._get_host_definitions_api()
+
+    def _get_dynamic_client(self):
+        return dynamic.DynamicClient(api_client.ApiClient(configuration=self._load_cluster_configuration()))
+
+    def _load_cluster_configuration(self):
+        return config.load_incluster_config()
+
+    def _get_csi_nodes_api(self):
+        return self.dynamic_client.resources.get(api_version=settings.STORAGE_API_VERSION,
+                                                 kind=settings.CSINODE_KIND)
+
+    def _get_host_definitions_api(self):
+        return self.dynamic_client.resources.get(api_version=settings.CSI_IBM_API_VERSION,
+                                                 kind=settings.HOST_DEFINITION_KIND)
+
+    def _get_csi_nodes_with_driver(self):
+        csi_nodes_with_driver = []
+        csi_nodes = self._get_csi_nodes()
+        for csi_node in csi_nodes:
+            if self._is_csi_node_has_driver(csi_node):
+                csi_nodes_with_driver.append(self._get_csi_node_object(csi_node))
+        return csi_nodes_with_driver
+
+    def _get_csi_nodes(self):
+        try:
+            return self.csi_nodes_api.get().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_CSI_NODES.format(ex.body))
+            return []
+
+    def _get_nodes(self):
+        try:
+            return self.core_api.list_node().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_NODES.format(ex.body))
+            return []
+
+    def _get_storage_classes(self):
+        try:
+            return self.storage_api.list_storage_class().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_STORAGE_CLASSES.format(ex.body))
+            return []
+
+    def _is_csi_node_has_driver(self, csi_node):
+        if csi_node.spec.drivers:
+            for driver in csi_node.spec.drivers:
+                return driver.name == settings.IBM_BLOCK_CSI_PROVISIONER_NAME
+        return False
+
+    def _get_csi_node(self, node_name):
+        try:
+            csi_node = self.csi_nodes_api.get(name=node_name)
+            return self._get_csi_node_object(csi_node)
+        except ApiException as ex:
+            if ex.status != 404:
+                logger.error(messages.CSI_NODE_DOES_NOT_EXIST.format(node_name))
+            else:
+                logger.error(messages.FAILED_TO_GET_CSI_NODE.format(node_name, ex.body))
+            return CsiNode()
+
+    def _get_csi_node_object(self, csi_node):
```

Comment:
`object` does not distinguish between the k8s and the internal csi node structures.
consider e.g. prefixing the k8s csi node term with `v1`/`k8s`/etc...

Reply:
I recommnad to rename the object CsiNodeInfo and then to reference it like it.. (`_get_csi_node_info`, `csi_node_info = CsiNodeInfo()` and get rid of object which is too general
same for HostDefinition

Reply:
also, I think _generate_csi_node_object is more aligned with our code.
same for any object initiators such as `_get_host_definition_object`, `_get_event_for_host_definition`..